airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -0,0 +1,562 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import copy
+import json
+import sys
+from datetime import datetime, timedelta, timezone
+from time import sleep
+from typing import TYPE_CHECKING, List, Optional, cast
+
+if sys.version_info >= (3, 8):
+    from functools import cached_property
+else:
+    from cached_property import cached_property
+
+from airflow.models import BaseOperator
+from airflow.operators.python import ShortCircuitOperator
+from airflow.providers.amazon.aws.hooks.appflow import AppflowHook
+
+if TYPE_CHECKING:
+    from mypy_boto3_appflow.client import AppflowClient
+    from mypy_boto3_appflow.type_defs import (
+        DescribeFlowExecutionRecordsResponseTypeDef,
+        ExecutionRecordTypeDef,
+        TaskTypeDef,
+    )
+
+    from airflow.utils.context import Context
+
+EVENTUAL_CONSISTENCY_OFFSET: int = 15  # seconds
+EVENTUAL_CONSISTENCY_POLLING: int = 10  # seconds
+SUPPORTED_SOURCES = {"salesforce", "zendesk"}
+
+
+class AppflowOperatorException(Exception):
+    """Alias for Exception."""
+
+
+class AppflowOperatorBase(BaseOperator):
+    """Amazon Appflow Base Operator class (not supposed to be used directly in DAGs)."""
+
+    BLUE = "#2bccbd"
+    ui_color = BLUE
+
+    def __init__(
+        self,
+        source: str,
+        name: str,
+        flow_update: bool,
+        source_field: Optional[str] = None,
+        dt: Optional[str] = None,
+        poll_interval: int = 20,
+        aws_conn_id: Optional[str] = "aws_default",
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        if source not in SUPPORTED_SOURCES:
+            raise AppflowOperatorException(
+                f"{source} is not a supported source (options: {SUPPORTED_SOURCES})!"
+            )
+        self.dt = dt
+        self._name = name
+        self._source = source
+        self._source_field = source_field
+        self._poll_interval = poll_interval
+        self._aws_conn_id = aws_conn_id
+        self._region = region
+        self._flow_update = flow_update
+
+    @cached_property
+    def hook(self) -> AppflowHook:
+        """Create and return an AppflowHook."""
+        return AppflowHook(aws_conn_id=self.aws_conn_id, region_name=self._region)
+
+    @staticmethod
+    def _dt_to_epoch_str(dt: datetime) -> str:
+        text = str(int(dt.timestamp() * 1000))
+        return text
+
+    def _get_connector_type(self) -> str:
+        connector_type = self._response["sourceFlowConfig"]["connectorType"]
+        if (self.source == "salesforce" and connector_type != "Salesforce") or (
+            self.source == "zendesk" and connector_type != "Zendesk"
+        ):
```

Comment:
Consider making this more future proof with `if self.source != connector_type.lower():`.  If you are also specifically trying to enforce that it's supported then perhaps `if self.source in SUPPORTED_SOURCES and self.source != connector_type.lower():`, But I don't think you need the `in` check here since you catch that in the init.

Reply:
Done!